core/templates/dev/head/domain/utilities/UrlInterpolationService.js

```diff
@@ -114,6 +114,108 @@ oppia.factory('UrlInterpolationService', ['alertsService',
         match = filledUrl.match(INTERPOLATION_VARIABLE_REGEX);
       }
       return filledUrl;
+    },
+
+    /**
+     * Given an resource path, returns the relative url path to that resource
+     * prefixing the appropriate cache_slug to it.
+     */
+    getStaticResourceUrl: function(resourcePath) {
+      if (!resourcePath) {
+        alertsService.fatalWarning(
+          'Empty resource path passed in getStaticResourceUrl.');
+        return null;
+      }
+
+      var RESOURCE_PATH_STARTS_WITH_FORWARD_SLASH = /^\//;
+      // Ensure that resourcePath starts with a forward slash.
+      if (!resourcePath.match(RESOURCE_PATH_STARTS_WITH_FORWARD_SLASH)) {
+        alertsService.fatalWarning(
+          'Resource path passed into getStaticResourceUrl must start with' +
+          '\'\/\': \'' + new String(resourcePath) + '\'.');
+        return null;
+      }
+
+      return GLOBALS.ASSET_DIR_PREFIX + resourcePath;
+    },
+
+    /**
+     * Given a domain url and resource path, returns the complete url path
+     * to that resource prefixing the appropriate cache_slug to it.
+     */
+    getCompleteStaticResourceUrl: function(domainUrl, resourcePath) {
+      if (!resourcePath) {
+        alertsService.fatalWarning(
+          'Empty resource path passed in getCompleteStaticResourceUrl.');
+        return null;
+      }
+
+      var RESOURCE_PATH_STARTS_WITH_FORWARD_SLASH = /^\//;
+      // Ensure that resourcePath starts with a forward slash.
+      if (!resourcePath.match(RESOURCE_PATH_STARTS_WITH_FORWARD_SLASH)) {
+        alertsService.fatalWarning(
+          'Resource path passed into getCompleteStaticResourceUrl must start' +
+          'with \'\/\': \'' + new String(resourcePath) + '\'.');
+        return null;
+      }
+
+      return domainUrl + GLOBALS.ASSET_DIR_PREFIX + resourcePath;
+    },
+
+    /**
+     * Given an image path, returns the complete url path to that image
+     * prefixing the appropriate cache_slug to it.
+     */
+    getStaticImageUrl: function(imagePath) {
+      if (!imagePath) {
+        alertsService.fatalWarning(
+          'Empty image path passed in getStaticImageUrl.');
+        return null;
+      }
+
+      var IMAGE_PATH_STARTS_WITH_FORWARD_SLASH = /^\//;
+      // Ensure that imagePath starts with a forward slash.
+      if (!imagePath.match(IMAGE_PATH_STARTS_WITH_FORWARD_SLASH)) {
+        alertsService.fatalWarning(
+          'Image path passed into getStaticImageUrl must start with \'\/\': ' +
+          '\'' + new String(imagePath) + '\'.');
+        return null;
+      }
+
+      return GLOBALS.ASSET_DIR_PREFIX + '/assets/images' + imagePath;
+    },
+
+    /**
+     * Given an extension resource path, returns the complete url path to that
+     * extension resource, prefixing the appropriate cache_slug to it.
+     */
+    getExtensionResourceUrl: function(extensionResourcePath) {
```

Comment:
I don't see it that way. The tests may look similar, but you're really testing different functions, which look independent from the point of view of someone viewing this service as a black box.

Basically, think about it from the point of view of a client of this service, who views the service as a black box (which is how you should be thinking when writing tests). The client doesn't know that the multiple functions which make up the external API make use of the same function internally. All they care about is that, when they call functions that are part of the external API, the resulting behaviour is correct. This is the point of the tests.